pyper/sink/__init__.py

```diff
@@ -1,5 +1,25 @@
-from pydantic.main import BaseModel
+from enum import Enum
+
+from pydantic import Field, BaseModel
+
+
+class SinkStatus(str, Enum):
+    NOT_STARTED = "not_started"
+    STARTED = "started"
+    ERROR = "error"
+    DONE = "done"
 
 
 class Sink(BaseModel):
-    ...
+    state: SinkStatus = Field(default=SinkStatus.NOT_STARTED)
+
+    def execute(self) -> None:
+        ...
```

Comment:
```suggestion
raise NotImplementedErorr("execute not implemented")
```

how about using NotImplementedError to abstract method